tests/test_modelrun.py

```diff
@@ -64,6 +68,22 @@ def test_builder(self, get_model_run_config_data):
         assert modelrun.narratives == config_data['narratives']
         assert modelrun.strategies == config_data['strategies']
 
+    def test_builder_scenario_sosmodelrun_not_in_sosmodel(self, get_model_run_config_data):
+
+        config_data = get_model_run_config_data
+        config_data['scenarios'] = {
+            'raininess': 'high_raininess',
+            'population': 'high_population'
+        }
+
+        builder = ModelRunBuilder()
+        builder.construct(config_data)
+
+        with raises(ModelRunError) as ex:
+            builder.finish()
+        assert "ScenarioSet 'population' is selected in the ModelRun " \
+               "configuration but not found in the SosModel configuration" in str(ex)
+
```

Comment:
Yes this check is already happening. Not a user-friendly error but the smif app should help users to avoid this situation as it should only be possible to select sosModelRun/scenarios that are exposed by the sosModel